modin/pandas/groupby.py

```diff
@@ -72,27 +74,33 @@ def __init__(
         self._by = by
         self._drop = drop
 
-        if (
-            level is None
-            and is_list_like(by)
-            or isinstance(by, type(self._query_compiler))
-        ):
-            # This tells us whether or not there are multiple columns/rows in the groupby
-            self._is_multi_by = (
-                isinstance(by, type(self._query_compiler)) and len(by.columns) > 1
-            ) or (
-                not isinstance(by, type(self._query_compiler))
-                and axis == 0
-                and all(
-                    (hashable(obj) and obj in self._query_compiler.columns)
-                    or isinstance(obj, type(self._query_compiler))
-                    or is_list_like(obj)
-                    for obj in self._by
-                )
-            )
+        if level is not None:
+            level = level if isinstance(level, list) else [level]
         else:
-            self._is_multi_by = False
-        self._level = level
+            level = []
+
+        # This tells us whether or not there are multiple columns/rows in the groupby
+        is_multi_column_qc = (
+            isinstance(by, type(self._query_compiler)) and len(by.columns) > 1
+        )
+        is_multi_by = (
+            is_list_like(by)
+            and not isinstance(by, type(self._query_compiler))
+            and axis == 0
+            and all(
+                is_label(self._df, obj)
+                or isinstance(obj, type(self._query_compiler))
+                or is_list_like(obj)
+                for obj in self._by
+            )
+        )
+        self._is_multi_by = (
+            is_multi_column_qc
+            or is_multi_by
+            or (len(level) > 0 and by is not None)
+            or len(level) > 1
+        )
```

Comment:
previous `is_multi_by` check considered only column names, now it also uses `level` parameter

Reply:
`is_multi_by` is only for multiple columns. Multiple levels is not an issue because each worker has all of the levels. `is_multi_by` indicates that we must broadcast multiple columns.

Reply:
got it, gonna revert this